WordPress/gutenberg
# Font Library Modal: Remove some contexts

Part of #58428

## What?

This PR removes the following context from the font library:

- `themeFonts`
- `baseThemeFonts`
- `customFonts`
- `fontFamiliesHasChanges`
- `hasResolvedLibrary` (This was never used in the first place)

## Why?

Aims to remove the context completely and trigger the font library modal via the command palette.

## How?

This PR is the first effort for #58428. I removed relatively simple processing from the context and implemented it ad hoc in each component.

## Testing Instructions

The font library behavior should be exactly the same as before, but in particular test the following:

- Ability to open and close font library modal
- Fonts can be uploaded and installed successfully.
